server.py

```diff
@@ -23,35 +31,63 @@ def return_request():
     while True:
         try:
             conn, addr = socket.accept()
-            output = ""
+            request = ""
             while True:
-                msg = conn.recv(16)
-                output = output + msg
-                if len(msg) < 16:
+                msg = conn.recv(1024)
+                request = request + msg
+                if len(msg) < 1024:
                     break
             try:
-                final = parse_request(output)
-                conn.sendall(response_ok(final))
+                response = response_ok(parse_request(request))
             except ValueError as detail:
-                detail = str(detail)
-                conn.sendall(response_error("400", detail))
+                response = response_error(b"400", str(detail))
+            except LookupError as detail:
+                response = response_error(b"404", str(detail))
+            except AttributeError as detail:
+                response = response_error(b"405", str(detail))
+            except NotImplementedError as detail:
+                response = response_error(b"500", str(detail))
+            except UserWarning as detail:
+                response = response_error(b"403", str(detail))
+            conn.sendall(response)
             conn.close()
         except KeyboardInterrupt:
             break
 
 
 def parse_request(request):
-    if "GET" not in request:
-        raise ValueError("Error.  Must send GET request")
-    if "HTTP/1.1" not in request:
-        raise ValueError("Error. Only accepts HTTP 1.1 Protocol")
-    if "Host:" not in request:
+    request = request.split("\r\n")
+    if "GET " not in request[0][:4]:
+        raise AttributeError("Error.  Must send GET request")
+    if "HTTP/1.1" not in request[0][-8:]:
+        raise NotImplementedError("Error. Only accepts HTTP 1.1 Protocol")
+    host = [o for o in request if o[:6] == "Host:"]
+    if host == []:
         raise ValueError("Error. Must provide host")
     else:
-        request_lines = request.split(b"\r\n")
-        first_line = request_lines[0].split(" ")
+        first_line = request[0].split(" ")
         return first_line[1]
 
 
+def resolve_uri(uri):
+    uri = os.path.join(ROOT_DIRECTORY + uri.strip(b"/"))
```

Comment:
be a bit careful here.  You don't want the URI to start with a '/', but you might want to preserve any that are at the end of the URI.